File: app/src/main/java/com/example/smartlight/TimeInterval.java
package com.example.smartlight;

import java.io.Serializable;

public class TimeInterval implements Serializable {
    //It needs to be serializabled to be able to serialize through different activities

    private int startHour;
    private int startMinute;
    private int endHour;
    private int endMinute;
    private int value;

    public TimeInterval(int startHour, int startMinute, int endHour, int endMinute, int value) {
        this.startHour = startHour;
        this.startMinute = startMinute;
        this.endHour = endHour;
        this.endMinute = endMinute;
        this.value = value;
    }

    public int getStartHour() {
        return startHour;
    }

    public void setStartHour(int startHour) {
        this.startHour = startHour;
    }

    public int getStartMinute() {
        return startMinute;
    }

    public void setStartMinute(int startMinute) {
        this.startMinute = startMinute;
    }

    public int getEndHour() {
        return endHour;
    }

    public void setEndHour(int endHour) {
        this.endHour = endHour;
    }

    public int getEndMinute() {
        return endMinute;
    }

    public void setEndMinute(int endMinute) {
        this.endMinute = endMinute;
    }

    public int getValue() {
        return value;
    }

    public void setValue(int value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return "TimeInterval{" + "startHour=" + startHour + ", startMinute=" + startMinute + ", endHour=" + endHour + ", endMinute=" + endMinute + ", value=" + value + '}';
    }
}

File: TimeInterval.py
class TimeInterval:
    def __init__(self, start_hour, start_minute, end_hour, end_minute, value):
        self.start_hour = start_hour
        self.start_minute = start_minute
        self.end_hour = end_hour
        self.end_minute = end_minute
        self.value = value

    def __str__(self):
        return f"TimeInterval{{start_hour={self.start_hour}, start_minute={self.start_minute}, " \
               f"end_hour={self.end_hour}, end_minute={self.end_minute}, value={self.value}}}"

    def get_start_hour(self):
        return self.start_hour

    def set_start_hour(self, start_hour):
        self.start_hour = start_hour

    def get_start_minute(self):
        return self.start_minute

    def set_start_minute(self, start_minute):
        self.start_minute = start_minute

    def get_end_hour(self):
        return self.end_hour

    def set_end_hour(self, end_hour):
        self.end_hour = end_hour

    def get_end_minute(self):
        return self.end_minute

    def set_end_minute(self, end_minute):
        self.end_minute = end_minute

    def get_value(self):
        return self.value

    def set_value(self, value):
        self.value = value